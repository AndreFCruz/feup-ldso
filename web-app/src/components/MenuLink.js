import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { Card, CardBody, CardTitle, Col } from 'reactstrap';
import PropTypes from 'prop-types';

class MenuLink extends Component {
    render() {
        return (
            <Col>
                <NavLink to={this.props.link}>
                    <Card>
                        <CardBody>
                            <CardTitle style={linkTextStyle}>
                                <p className="center-text" style={title}>Manage {this.props.title}</p>
                            </CardTitle>
                        </CardBody>
                    </Card>
                </NavLink >
            </Col>
        );
    }
}

const linkTextStyle = {
    minHeight: '400px',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center'
};

const title = {
    fontSize: '2rem',
    fontWeight: 700,
    color: 'rgb(33, 37, 41)'
};

MenuLink.propTypes = {
    link: PropTypes.string,
    title: PropTypes.string
};

export default MenuLink;
