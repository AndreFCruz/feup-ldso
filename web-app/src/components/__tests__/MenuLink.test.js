import React from 'react';
import MenuLink from '../MenuLink';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('renders correct link', () => {
    const wrapper = shallow(<MenuLink link={'/faculties/FEUP/projects'} title={'Social Impact Projects'} />);

    expect(wrapper).toMatchSnapshot();
});