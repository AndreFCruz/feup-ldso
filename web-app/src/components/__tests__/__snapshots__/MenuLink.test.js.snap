// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders without crashing 1`] = `
<div>
  <div
    className="loginContainer"
  >
    <div
      className="cssTool"
    />
    <div
      className="login"
    >
      <div
        className="logo"
      >
        <img
          alt="application logo"
          src="test-file-stub"
          width="140"
        />
      </div>
      <form
        className="form"
      >
        <div
          className="col"
        >
          <div
            className="position-relative form-group"
          >
            <label
              className=""
              htmlFor="username"
            >
              Username
            </label>
            <input
              className="form-control"
              id="username"
              name="username"
              onChange={[Function]}
              placeholder="Enter Username"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="position-relative form-group"
          >
            <label
              className=""
              htmlFor="username"
            >
              Password
            </label>
            <input
              className="form-control"
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
          </div>
          <button
            aria-label={null}
            className="loginBtn  btn btn-success"
            onClick={[Function]}
            type="button"
          >
            Log in
          </button>
        </div>
      </form>
    </div>
  </div>
  <div
    className="notification-container notification-container-empty"
  >
    <span />
  </div>
</div>
`;

exports[`renders correct link 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuLink
    link="/faculties/FEUP/projects"
    title="Social Impact Projects"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <NavLink
        activeClassName="active"
        aria-current="page"
        to="/faculties/FEUP/projects"
      >
        <Card
          tag="div"
        >
          <CardBody
            tag="div"
          >
            <CardTitle
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                  "minHeight": "400px",
                }
              }
              tag="h5"
            >
              <p
                className="center-text"
                style={
                  Object {
                    "color": "rgb(33, 37, 41)",
                    "fontSize": "2rem",
                    "fontWeight": 700,
                  }
                }
              >
                Manage 
                Social Impact Projects
              </p>
            </CardTitle>
          </CardBody>
        </Card>
      </NavLink>,
      "tag": "div",
      "widths": Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "activeClassName": "active",
        "aria-current": "page",
        "children": <Card
          tag="div"
        >
          <CardBody
            tag="div"
          >
            <CardTitle
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                  "minHeight": "400px",
                }
              }
              tag="h5"
            >
              <p
                className="center-text"
                style={
                  Object {
                    "color": "rgb(33, 37, 41)",
                    "fontSize": "2rem",
                    "fontWeight": 700,
                  }
                }
              >
                Manage 
                Social Impact Projects
              </p>
            </CardTitle>
          </CardBody>
        </Card>,
        "to": "/faculties/FEUP/projects",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <CardBody
            tag="div"
          >
            <CardTitle
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                  "minHeight": "400px",
                }
              }
              tag="h5"
            >
              <p
                className="center-text"
                style={
                  Object {
                    "color": "rgb(33, 37, 41)",
                    "fontSize": "2rem",
                    "fontWeight": 700,
                  }
                }
              >
                Manage 
                Social Impact Projects
              </p>
            </CardTitle>
          </CardBody>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <CardTitle
              style={
                Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                  "minHeight": "400px",
                }
              }
              tag="h5"
            >
              <p
                className="center-text"
                style={
                  Object {
                    "color": "rgb(33, 37, 41)",
                    "fontSize": "2rem",
                    "fontWeight": 700,
                  }
                }
              >
                Manage 
                Social Impact Projects
              </p>
            </CardTitle>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <p
                className="center-text"
                style={
                  Object {
                    "color": "rgb(33, 37, 41)",
                    "fontSize": "2rem",
                    "fontWeight": 700,
                  }
                }
              >
                Manage 
                Social Impact Projects
              </p>,
              "style": Object {
                "alignItems": "center",
                "display": "flex",
                "justifyContent": "center",
                "minHeight": "400px",
              },
              "tag": "h5",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Manage ",
                  "Social Impact Projects",
                ],
                "className": "center-text",
                "style": Object {
                  "color": "rgb(33, 37, 41)",
                  "fontSize": "2rem",
                  "fontWeight": 700,
                },
              },
              "ref": null,
              "rendered": Array [
                "Manage ",
                "Social Impact Projects",
              ],
              "type": "p",
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <NavLink
          activeClassName="active"
          aria-current="page"
          to="/faculties/FEUP/projects"
        >
          <Card
            tag="div"
          >
            <CardBody
              tag="div"
            >
              <CardTitle
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "justifyContent": "center",
                    "minHeight": "400px",
                  }
                }
                tag="h5"
              >
                <p
                  className="center-text"
                  style={
                    Object {
                      "color": "rgb(33, 37, 41)",
                      "fontSize": "2rem",
                      "fontWeight": 700,
                    }
                  }
                >
                  Manage 
                  Social Impact Projects
                </p>
              </CardTitle>
            </CardBody>
          </Card>
        </NavLink>,
        "tag": "div",
        "widths": Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "aria-current": "page",
          "children": <Card
            tag="div"
          >
            <CardBody
              tag="div"
            >
              <CardTitle
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "justifyContent": "center",
                    "minHeight": "400px",
                  }
                }
                tag="h5"
              >
                <p
                  className="center-text"
                  style={
                    Object {
                      "color": "rgb(33, 37, 41)",
                      "fontSize": "2rem",
                      "fontWeight": 700,
                    }
                  }
                >
                  Manage 
                  Social Impact Projects
                </p>
              </CardTitle>
            </CardBody>
          </Card>,
          "to": "/faculties/FEUP/projects",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <CardBody
              tag="div"
            >
              <CardTitle
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "justifyContent": "center",
                    "minHeight": "400px",
                  }
                }
                tag="h5"
              >
                <p
                  className="center-text"
                  style={
                    Object {
                      "color": "rgb(33, 37, 41)",
                      "fontSize": "2rem",
                      "fontWeight": 700,
                    }
                  }
                >
                  Manage 
                  Social Impact Projects
                </p>
              </CardTitle>
            </CardBody>,
            "tag": "div",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <CardTitle
                style={
                  Object {
                    "alignItems": "center",
                    "display": "flex",
                    "justifyContent": "center",
                    "minHeight": "400px",
                  }
                }
                tag="h5"
              >
                <p
                  className="center-text"
                  style={
                    Object {
                      "color": "rgb(33, 37, 41)",
                      "fontSize": "2rem",
                      "fontWeight": 700,
                    }
                  }
                >
                  Manage 
                  Social Impact Projects
                </p>
              </CardTitle>,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <p
                  className="center-text"
                  style={
                    Object {
                      "color": "rgb(33, 37, 41)",
                      "fontSize": "2rem",
                      "fontWeight": 700,
                    }
                  }
                >
                  Manage 
                  Social Impact Projects
                </p>,
                "style": Object {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                  "minHeight": "400px",
                },
                "tag": "h5",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Manage ",
                    "Social Impact Projects",
                  ],
                  "className": "center-text",
                  "style": Object {
                    "color": "rgb(33, 37, 41)",
                    "fontSize": "2rem",
                    "fontWeight": 700,
                  },
                },
                "ref": null,
                "rendered": Array [
                  "Manage ",
                  "Social Impact Projects",
                ],
                "type": "p",
              },
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
